### PR TITLE
Remove poise_service and poise_archive dependency

### PR DESCRIPTION
**Pull request by zelaskov**
- Removed poise_service dependency, used Chef `service` resource
- Creation of user and group by using Chef `user` and `group` resource
- Change creation of config file - removed agent.yaml and changed it to newrelic.yml to be the same as it is in https://github.com/newrelic/infrastructure-agent-puppet or https://github.com/newrelic/infrastructure-agent-ansible
- *_repositories section in agent_linux rewritten as there were issues with chef 17 +
- fixed tests
- updated kitchen.yml platforms to better represent [requirements for the infra agent](https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/get-started/requirements-infrastructure-agent/)
- tested on all platforms in kitchen.yml with chef 16 and 17 and everything seems fine :)

### CODE REVIEW

File: libraries/integration.rb [hunk 1]
```diff
@@ -90,7 +90,7 @@ def config_file_content
       end
 
       # Fetch the remote executable tarball if the install method is set to `tarball`
-      poise_archive new_resource.remote_url do
+      archive_file new_resource.remote_url do
```
Comment (varas) [inline libraries/integration.rb:93]: `archive_file` is available from 15+ https://docs.chef.io/resources/archive_file/
And we provide support for 12.15+ as per: https://github.com/newrelic/infrastructure-agent-chef#chef

File: recipes/agent_linux.rb [hunk 2]
```diff
@@ -13,10 +13,14 @@
...
 )
 
+group node['newrelic_infra']['group']['name'] do
+  group_name node['newrelic_infra']['group']['name']
+end
+
 # Setup a service account
-poise_service_user node['newrelic_infra']['user']['name'] do
-  group node['newrelic_infra']['group']['name']
-  only_if { node['newrelic_infra']['features']['manage_service_account'] }
```
Comment (varas) [inline recipes/agent_linux.rb:19]: I guess we are removing this flag, but I don't see it removed:
- neither at [docs](https://github.com/newrelic/infrastructure-agent-chef#attributes)
- nor at [default](https://github.com/newrelic/infrastructure-agent-chef/blob/fd9aa6ffb0c3985451877f941eec7a83d86dad44/attributes/default.rb#L21-L22)

Comment (zelaskov) [reply recipes/agent_linux.rb:19]: I removed flag and updated docs

**Comment by zelaskov** [comment]
@varas Hello! I've made changes to docs and I removed unused flag.

**Review by varas — APPROVED** [review]

**Pull request merged**